preflight_check/utils.py

```diff
@@ -0,0 +1,234 @@
+    def fence_action(self):
+        action_result = get_property("stonith-action")
+        if action_result is None or action_result not in ["off", "poweroff", "reboot"]:
+            msg_error("Cluster property \"stonith-action\" should be reboot|off|poweroff")
+            return None
+        return action_result
+
+    @property
+    def fence_timeout(self):
+        timeout_result = get_property("stonith-timeout")
+        if timeout_result and re.match(r'[1-9][0-9]*(s|)$', timeout_result):
+            return timeout_result.strip("s")
+        return config.FENCE_TIMEOUT
+
+
+def check_node_status(node, state):
+    """
+    Check whether the node has expected state
+    """
+    rc, stdout, stderr = crmshutils.get_stdout_stderr('crm_node -l')
+    if rc != 0:
+        msg_error(stderr)
+        return False
+    pattern = re.compile(r'^.* {} {}'.format(node, state), re.MULTILINE)
+    if not pattern.search(stdout):
+        return False
+    return True
+
+
+def online_nodes():
+    """
+    Get online node list
+    """
+    rc, stdout, stderr = crmshutils.get_stdout_stderr('crm_mon -1')
+    if rc == 0 and stdout:
+        res = re.search(r'Online:\s+\[\s(.*)\s\]', stdout)
+        if res:
+            return res.group(1).split()
+    return []
+
+
+def peer_node_list():
+    """
+    Get online node list except self
+    """
+    online_nodelist = online_nodes()
+    if online_nodelist:
+        online_nodelist.remove(this_node())
+        return online_nodelist
+    return []
+
+
+def this_node():
+    """
+    Try to get the node name from crm_node command
+    If failed, use its hostname
+    """
+    rc, stdout, stderr = crmshutils.get_stdout_stderr("crm_node --name")
+    if rc != 0:
+        msg_error(stderr)
+        return crmshutils.this_node()
+    return stdout
+
+
+def str_to_datetime(str_time, fmt):
+    return datetime.strptime(str_time, fmt)
+
+
+def corosync_port_list():
+    """
+    Get corosync ports using corosync-cmapctl
+    """
+    ports = []
+    rc, out, _ = crmshutils.get_stdout_stderr("corosync-cmapctl totem.interface")
+    if rc == 0 and out:
+        ports = re.findall(r'(?:mcastport.*) ([0-9]+)', out)
+    return ports
+
+
+def get_handler(logger, _type):
+    """
+    Get logger specific handler
+    """
+    for h in logger.handlers:
+        if getattr(h, '_name') == _type:
+            return h
+
+
+def is_root():
+    return os.getuid() == 0
+
+
+def get_process_status(s):
+    """
+    Returns true if argument is the name of a running process.
+
+    s: process name
+    returns Boolean and pid
+    """
+    # find pids of running processes
+    pids = [pid for pid in os.listdir('/proc') if pid.isdigit()]
+    for pid in pids:
+        try:
+            pid_file = os.path.join('/proc', pid, 'cmdline')
+            with open(pid_file, 'rb') as f:
+                data = f.read()
+                procname = os.path.basename(crmshutils.to_ascii(data).replace('\x00', ' ').split(' ')[0])
```

Comment:
How about use library `psutil.process_iter()` or even using shell command `pidof` to replace path through the `/proc`?

Reply:
We have similar discuss before https://github.com/ClusterLabs/crmsh/pull/185
Anyway, current code works well